### ISSUE TITLE
MQTT error

### ISSUE THREAD

**Issue by TheRealFalseReality** [reef-pi/reef-pi#2009]
I see 2 other posts regarding MQTT errors, but haven't seen any progress on them. I use Home Assistant with Mosquitto broker, and use other mqtt clients no issue, using the credentials needed. I do use the Reef-Pi HA Integration with no issue, but I would also like MQTT if possible. 

I don't know if I am entering the wrong information in ReefPi or something. I attached screenshots of my config in hoped someone can point out a flaw. 
Would it be a ReefPi bug or something on my end? I get the 
Failed to initialize mqtt client:network Error : dial tcp: lookup homeassistant.local on
issue when retrying the URLs

![Screenshot 2023-01-12 161324](https://user-images.githubusercontent.com/106857076/212183020-2d99bf45-b98b-443b-bdc9-8bfadae3efc0.png)
![Screenshot 2023-01-12 160908](https://user-images.githubusercontent.com/106857076/212183029-caa76f8a-dba5-4941-a477-2bf4aebdfc91.png)

**Comment by TheStaches** [comment]
Have you tried using the direct IP address to your Home Assistant service? Your field inputs look good, assuming your user and password are valid. 

You may need to reload reefpi  after updating MQTT info. 

You can also try using the retained checkbox. I don't know what it does but was trying it when diagnosing my own connection.

**Comment by TheRealFalseReality** [comment]
Oh yes, I have. With the hostname, I get that above error with the stuff in brackets.

**Comment by TheStaches** [comment]
Looks like progress. 

It looks like the connection is available but failing on the credentials. 

Try installing mqtt-explorer in Home Assistant and setting up the connection to your mosquito broker. Once setup, it will be the same credentials used in reefpi.

http://mqtt-explorer.com/

**Comment by TheRealFalseReality** [comment]
Ok, I'll give that a try

> Looks like progress.
> 
> It looks like the connection is available but failing on the credentials.
> 
> Try installing mqtt-explorer in Home Assistant and setting up the connection to your mosquito broker. Once setup, it will be the same credentials used in reefpi.
> 
> http://mqtt-explorer.com/

**Comment by TheRealFalseReality** [comment]
It's spitting out the IPv6 address for a DNS server and port 53. Unsure what this means so far

![Screenshot 2023-01-14 131748](https://user-images.githubusercontent.com/106857076/212488961-cbdfebd6-538a-4320-88cc-7191e6e67153.png)


> Looks like progress.
> 
> It looks like the connection is available but failing on the credentials.
> 
> Try installing mqtt-explorer in Home Assistant and setting up the connection to your mosquito broker. Once setup, it will be the same credentials used in reefpi.
> 
> http://mqtt-explorer.com/

**Comment by TheStaches** [comment]
Hmmm, not sure what that page is referencing. 

mqtt-explorer should have a web UI that you can access. This is where you can input your credentials. 

Setting -> Addons -> mqtt-explorer -> Open Web UI

**Comment by TheRealFalseReality** [comment]
I'm never used MQTT Explorer before now, but got connected to my HA, I see see a binary sensor for "Display" labeled as ReefPi, could that be something?

![Screenshot 2023-01-15 191256](https://user-images.githubusercontent.com/106857076/212575189-ecf1bc16-e9fa-4c49-99c6-a144ca442904.png)


> Hmmm, not sure what that page is referencing.
> 
> mqtt-explorer should have a web UI that you can access. This is where you can input your credentials.
> 
> Setting -> Addons -> mqtt-explorer -> Open Web UI

**Comment by TheStaches** [comment]
Based on where it is the data tree, I would guess it might be from the Reef-Pi HA integration you are using. 

MQTT messages from reef-pi should start at the same level as your octopi messages. 

Have you tried the same credentials you used to connect with mqtt-explorer, to connect with reef pi?

**Comment by TheRealFalseReality** [comment]
Yep, using same credentials and URL as all my other MQTT enabled devices. Only having trouble with Reef-Pi... Not a major issue, but I would like the extra telemetry. 

> Based on where it is the data tree, I would guess it might be from the Reef-Pi HA integration you are using.
> 
> MQTT messages from reef-pi should start at the same level as your octopi messages.
> 
> Have you tried the same credentials you used to connect with mqtt-explorer, to connect with reef pi?

**Comment by TheStaches** [comment]
That is odd. The only other things that I can think of are to reload/reboot reef-pi. I know you need to reload the UI after saving credentials. 

I'm running version 5.3. 

That's about the extent of my knowledge, sorry we couldn't get this working for you.
